ec/src/hashing/curve_maps/wb/mod.rs

```diff
@@ -13,45 +13,69 @@ use crate::{
 use super::swu::{SWUMap, SWUParams};
 type BaseField<MP> = <MP as CurveConfig>::BaseField;
 
-/// Trait defining the necessary parameters for the WB hash-to-curve method
-/// for the curves of Weierstrass form of:
-/// of y^2 = x^3 + a*x + b where b != 0 but `a` can be zero like BLS-381 curve.
-/// From [\[WB2019\]]
+/// [`IsogenyMap`] defines an isogeny between curves of
+/// form `Phi(x, y) := (a(x), b(x)*y).
+/// The `x` coordinate of the codomain point only depends on the
+/// `x`-coordinate of the domain point, and the
+/// `y`-coordinate of the codomain point is a multiple of the `y`-coordinate of the domain point.
+/// The multiplier depends on the `x`-coordinate of the domain point.
+/// All isogeny maps of curves of short Weierstrass form can be written in this way. See
+/// [\[Ga18]\]. Theorem 9.7.5 for details.
 ///
-/// - [\[WB2019\]] <http://dx.doi.org/10.46586/tches.v2019.i4.154-179>
-pub trait WBParams: SWCurveConfig + Sized {
-    // The isogenous curve should be defined over the same base field but it can have
-    // different scalar field type IsogenousCurveScalarField :
-    type IsogenousCurve: SWUParams<BaseField = BaseField<Self>>;
-
-    const PHI_X_NOM: &'static [BaseField<Self>];
-    const PHI_X_DEN: &'static [BaseField<Self>];
-
-    const PHI_Y_NOM: &'static [BaseField<Self>];
-    const PHI_Y_DEN: &'static [BaseField<Self>];
+/// - [\[Ga18]\] Galbraith, S. D. (2018). Mathematics of public key cryptography.
+pub struct IsogenyMap<
+    'a,
+    DOMAINE: SWCurveConfig,
+    CODOMAINE: SWCurveConfig<BaseField = BaseField<DOMAINE>>,
+> {
+    pub x_map_numerator: &'a [BaseField<CODOMAINE>],
+    pub x_map_denominator: &'a [BaseField<CODOMAINE>],
+
+    pub y_map_numerator: &'a [BaseField<CODOMAINE>],
+    pub y_map_denominator: &'a [BaseField<CODOMAINE>],
+
+    pub _phantom_domain: PhantomData<DOMAINE>,
+}
```

Comment:
```suggestion
pub struct IsogenyMap<
    'a,
    DOMAIN: SWCurveConfig,
    CODOMAIN: SWCurveConfig<BaseField = BaseField<DOMAIN>>,
> {
    pub x_map_numerator: &'a [BaseField<CODOMAIN>],
    pub x_map_denominator: &'a [BaseField<CODOMAIN>],

    pub y_map_numerator: &'a [BaseField<CODOMAIN>],
    pub y_map_denominator: &'a [BaseField<CODOMAIN>],
}
```

Reply:
Ah, sorry, didn't realize that we we weren't using `DOMAIN`. There's two options:
* Reinstate the `PhantomData<CODOMAIN>` field, but make it private. To enable constructing an `IsogenyMap`, let's add a `new` method which does the construction.
* Change the `*_map_denominator` type signatures to be `&'a [BaseField<DOMAIN>]`. Admittedly this is a bit of a hack, but this avoids adding back the field.

Reply:
> * To enable constructing an `IsogenyMap`, let's add a `new` method which does the construction.

Hmm, then new need to be a const function

Reply:
That should be possible, since it's a simple method, no?

Reply:
Maybe I'll go with the second solution then.

Reply:
We only need the method to be a `const fn`; the arguments don't have to be explicitly `const`; by calling `new` in a const context, the arguments are constrained to be `const`

Reply:
hmm but they need to have static lifetime no?

Reply:
I try sending them as array instead of refrences...

Reply:
I can't because we don't know they don't have the same size across the curves.

Reply:
@Pratyush [here](https://github.com/arkworks-rs/algebra/pull/517) is the code with new constructor. If you tell me how to send the slices with static lifetime to new function so it compiles, then I'll update this PR accordingly.